boards: st: add initial support for the stm32h757i eval board

tgcfoss opened the pull request:
Add initial support for the stm32h757i eval board.
Largely based on the stm32h747 disco board entry.

nordicjm commented on boards/shields/st_b_lcd40_dsi1_mb1166/boards/stm32h757i_eval_stm32h757xx_m7.defconfig:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2024 Linumiz
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_STM32H757I_EVAL_STM32H757XX_M7
+if LVGL
+
```
```suggestion
if BOARD_STM32H757I_EVAL_STM32H757XX_M7

if LVGL

```

tgcfoss replied:
fixed

nordicjm commented on boards/shields/st_b_lcd40_dsi1_mb1166/boards/stm32h757i_eval_stm32h757xx_m7.defconfig:
```diff
@@ -0,0 +1,21 @@
+config LV_Z_FULL_REFRESH
+	default y
+
+endif #LVGL
+endif # BOARD_STM32H757I_EVAL_STM32H757XX_M7
```
```suggestion
endif # LVGL

endif # BOARD_STM32H757I_EVAL_STM32H757XX_M7
```

tgcfoss replied:
changed as suggested

nordicjm commented on boards/st/stm32h757i_eval/board.cmake:
```diff
@@ -0,0 +1,26 @@
+
+# keep first
+if(CONFIG_STM32_MEMMAP)
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
```
2-space indent

nordicjm commented on boards/st/stm32h757i_eval/dc_ram.ld:
```diff
@@ -0,0 +1,16 @@
+ */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(sdram2), okay)
+GROUP_START(SDRAM2)
```
this isn't going to work for people using a cmake generator @tejlmand

tgcfoss replied:
Is this something I should address? And if so, how? This part was "inherited" from the stm32h747i-disco board.

nordicjm commented on boards/st/stm32h757i_eval/doc/img/stm32h757i_eval.jpg:
webp is likely smaller

tgcfoss replied:
Replaced with webp file

nordicjm commented on boards/st/stm32h757i_eval/doc/index.rst:
```diff
@@ -0,0 +1,298 @@
+Supported Features
+==================
+
+The current Zephyr stm32h757i_eval board configuration supports the following hardware features:
```
```suggestion
The Zephyr ``stm32h757i_eval`` board supports the following hardware features:
```

tgcfoss replied:
changed as suggested

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval.dtsi:
```diff
@@ -0,0 +1,118 @@
+/ {
+	leds {
+		compatible = "gpio-leds";
+		green_led_1:led_1 {
```
missing newline spacing throughout file with child nodes

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval.dtsi:
```diff
@@ -0,0 +1,118 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//#include "arduino_r3_connector.dtsi"
+//#include "pmod_connector.dtsi"
```
commented lines can go

tgcfoss replied:
removed

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m4.dts:
```diff
@@ -0,0 +1,60 @@
+		green_led_1:led_1 {
+			status = "okay";
+		};
+		orange_led_2:led_2 {
```
as above

tgcfoss replied:
changed as suggested

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m7.dts:
```diff
@@ -0,0 +1,298 @@
+		green_led_1:led_1 {
+			status = "okay";
+		};
+		orange_led_2:led_2 {
```
as above

tgcfoss replied:
changed as suggested

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m7.dts:
```diff
@@ -0,0 +1,298 @@
+	};
+};
+
+//arduino_spi: &spi5 {};
```
can go?

tgcfoss replied:
removed

miggazElquez commented:
You should squash your commits (https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#pr-requirements)

tgcfoss committed:
boards: st: add initial support for the stm32h757i eval board

Add initial support for the STM32H757I-EVAL board.
Largely based on the STM32H747I-DISCO board

Signed-off-by: Tomas Groth Christensen <tgc@foss.dk>

kartben commented on boards/st/stm32h757i_eval/doc/index.rst:
```diff
@@ -0,0 +1,298 @@
+
+The Zephyr  ``stm32h757i_eval`` board supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| ETHERNET  | on-chip    | ethernet                            |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+| FMC       | on-chip    | memc (SDRAM)                        |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| QSPI NOR  | on-chip    | off-chip flash                      |
++-----------+------------+-------------------------------------+
+| SDMMC     | on-chip    | disk access                         |
++-----------+------------+-------------------------------------+
+| IPM       | on-chip    | virtual mailbox based on HSEM       |
++-----------+------------+-------------------------------------+
+| DISPLAY   | on-chip    | MIPI DSI Host with shield (MP1166)  |
+|           |            | st_b_lcd40_dsi1_mb1166              |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+The default configuration per core can be found in the defconfig files:
+:zephyr_file:`boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m7_defconfig` and
+:zephyr_file:`boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m4_defconfig`
```
Please replace with the following and let magic fill out the table for you :) (see e.g. https://docs.zephyrproject.org/latest/boards/st/stm32h747i_disco/doc/index.html#supported-features)

```suggestion
.. zephyr:board-supported-hw::
```

kartben reviewed (changes requested):
Thanks for the PR! Just spotted a minor request for change.

erwango commented on boards/st/stm32h757i_eval/board.cmake:
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2019 STMicroelectronics
+
+# keep first
+if(CONFIG_STM32_MEMMAP)
+  board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
+  # not supported for STM32H757I-EVAL (?)
+  #board_runner_args(stm32cubeprogrammer "--extload=MT25TL01G_STM32H757I-EVAL.stldr")
```
If this doesn't work (yet), just remove the case.

tgcfoss replied:
Removed

erwango commented on boards/shields/st_b_lcd40_dsi1_mb1166/boards/st_b_lcd40_dsi1_mb1166/stm32h757i_eval_stm32h757xx_m7.conf:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2023 bytes at work AG
+# Copyright (c) 2024 Erik Andersson <erian747@gmail.com>
```
Please use your own copyright. Apply everywhere

tgcfoss replied:
Done

erwango commented on boards/st/stm32h757i_eval/stm32h757i_eval.dtsi:
```diff
@@ -0,0 +1,127 @@
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <4 0 &gpiob 14 0>,	// TOUCH_INT
```
```suggestion
		gpio-map = <4 0 &gpiob 14 0>,	/* TOUCH_INT */
```
Apply everywhere

tgcfoss replied:
Fixed

erwango commented on boards/st/stm32h757i_eval/stm32h757i_eval.dtsi:
```diff
@@ -0,0 +1,127 @@
+	current-speed = <115200>;
+};
+
+&mailbox {
+	status = "okay";
+};
```
Have you been able to test this ?
In any case, only enable what you've actually been able to to verify.

tgcfoss replied:
Removed

kartben reviewed (dismissed):
+1 for docs

erwango commented on boards/st/stm32h757i_eval/doc/index.rst:
```diff
@@ -0,0 +1,260 @@
+.. _STM32CubeProgrammer:
+   https://www.st.com/en/development-tools/stm32cubeprog.html
+
+.. _DISCO_H747I modifications for Ethernet:
+   https://os.mbed.com/teams/ST/wiki/DISCO_H747I-modifications-for-Ethernet
```
To be removed

tgcfoss replied:
Removed, good catch.

erwango reviewed (changes requested):
Otherwise LGTM

nordicjm commented on boards/shields/st_b_lcd40_dsi1_mb1166/boards/stm32h757i_eval_stm32h757xx_m7.defconfig:
```diff
@@ -0,0 +1,24 @@
+config LV_Z_FULL_REFRESH
+	default y
+
+endif #LVGL
```
```suggestion
endif # LVGL
```
This was requested before and seems to have been reverted (by mistake?) https://github.com/zephyrproject-rtos/zephyr/pull/86610#discussion_r1979147130

tgcfoss replied:
fixed

kartben commented:
@tgcfoss you have a minor compliance error left to fix

tgcfoss commented:
> @tgcfoss you have a minor compliance error left to fix

Fixed

tgcfoss commented on boards/shields/st_b_lcd40_dsi1_mb1166/boards/stm32h757i_eval_stm32h757xx_m7.conf:
```diff
@@ -0,0 +1,11 @@
+
+CONFIG_MEMC=y
+CONFIG_STM32_LTDC_ARGB8888=y
+CONFIG_HEAP_MEM_POOL_SIZE=65536
```
This was something inherited from the stm32h7-disco board. Did a simple test and it seems it was not needed, so removed it.

erwango reviewed (changes requested):
Please fix reported build failures

kartben commented:
> Please fix reported build failures

@erwango You're the assignee here, so it doesn't make a big difference, but please keep [this](https://docs.zephyrproject.org/latest/contribute/reviewer_expectations.html#reviewer-expectations:~:text=Reviewers%20should%20not%20request%20changes%20for%20issues%20which%20are%20automatically%20caught%20by%20CI%2C%20as%20this%20causes%20the%20pull%20request%20to%20remain%20blocked%20even%20after%20CI%20failures%20have%20been%20addressed%20and%20may%20unnecessarily%20delay%20it%20from%20being%20merged) in mind :)

erwango commented on boards/st/stm32h757i_eval/CMakeLists.txt:
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright 2023 BrainCo Inc.
```
# Copyright (c) 2023 BrainCo Inc.

tgcfoss replied:
Fixed

tgcfoss commented on boards/st/stm32h747i_disco/stm32h747i_disco_stm32h747xx_m7.dts:
```diff
@@ -127,14 +127,6 @@
-	 * microphone.
-	 * Cf Ethernet section in board documentation for more information on
-	 * the hw modification to be done to enable it.
-	 */
```
This was committed by error, I've reverted it.

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval_stm32h757xx_m4.dts:
```diff
@@ -0,0 +1,64 @@
+	};
+
+	leds {
+		green_led_1:led_1 {
```
needs spaces after the colon i.e. `: ` for whole PR

tgcfoss replied:
fixed

nordicjm commented on boards/st/stm32h757i_eval/stm32h757i_eval.dtsi:
```diff
@@ -0,0 +1,124 @@
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map =	<4 0 &gpiob 14 0>,	/* TOUCH_INT */
+					<35 0 &gpioe 5 0>,	/* SCLK/MCLK */
+					<37 0 &gpioe 4 0>,	/* LRCLK */
+					<39 0 &gpioe 6 0>,	/* I2S */
+					<40 0 &gpiob 7 0>,	/* I2C4_SDA (=>I2C1_SDA) */
+					<44 0 &gpiod 12 0>,	/* I2C4_SCL (=>I2C1_SCL) */
+					<45 0 &gpioa 8 0>,	/* CEC_CLK */
+					<47 0 &gpioa 15 0>,	/* CEC */
+					<49 0 &gpioj 2 0>,	/* DSI_TE */
+					<53 0 &gpioa 6 0>,	/* LCD_BL_CTRL */
+					<57 0 &gpiof 10 0>;	/* DSI_RESET */
```
should be one space after =, then align rest with that using tabs and spaces
